File: haskell-src/D1Bin.hs
import D1Lib
import qualified System.IO

inputFile :: String
inputFile = "input/d1.txt"

main = do
    contents <- System.IO.readFile inputFile
    let part1Out = part1 contents
        part2Out = part2 contents
    putStrLn $ "p1: " ++ show part1Out
    putStrLn $ "p2: " ++ show part2Out

File: rust-src/d1bin.rs
mod d1lib;

const INPUT: &str = "input/d1.txt";

fn main() {
    let input = std::fs::read_to_string(INPUT).unwrap();
    println!("p1: {}", d1lib::part1(&input));
    println!("p2: {}", d1lib::part2(&input));
}
